client/store/cart.js

```diff
@@ -98,3 +110,71 @@ export default function(state = initialState, action) {
       return state
   }
 }
+
+export function shouldPromptUser(state = false, action) {
+  switch (action.type) {
+    case SET_PROMPT_USER_ADD_LOCAL_CART:
+      return action.decision
+    default:
+      return state
+  }
+}
+
+export function localCartMiddleware(store) {
+  return next => action => {
+    if (action.type === CHECK_LOCALSTORAGE) {
+      let state = store.getState()
```

Comment:
This isn't a thunk

why not use currying and return a function that passes dispatch as a parameter  like the rest of the codebase?

Reply:
it would make everything look more consistent. this sticks out as having a way different style as the rest of the code